generators/csharp/codegen/src/ast/Type.ts

```diff
@@ -154,7 +154,9 @@ export class Type extends AstNode {
                 break;
             case "optional":
                 this.internalType.value.write(writer);
-                writer.write("?");
+                if (writer.getLastCharacter() !== "?") {
```

Comment:
curious what you think of this maybe hacky approach @dsinghvi

Reply:
interesting -- can we pass an optional parent type into this func instead? Feels like this will break pretty soon

Reply:
worked!